sqlmesh/core/config/connection.py

```diff
@@ -148,199 +150,139 @@ def _connection_factory(self) -> t.Callable:
-    _http_headers_validator = http_headers_validator
-
-    _has_spark_session_access: bool
-
-    class Config:
-        allow_mutation = True
-
-    @property
-    def has_spark_session_access(self) -> bool:
-        if not getattr(self, "_has_spark_session_access", None):
-            try:
-                from pyspark.sql import SparkSession
-
-                spark = SparkSession.getActiveSession()
-                if spark:
-                    self._has_spark_session_access = True
-                    self.concurrent_tasks = 1
-                else:
-                    self._has_spark_session_access = False
-            except ImportError:
-                self._has_spark_session_access = False
-        return self._has_spark_session_access
+    def _engine_adapter(self) -> t.Type[engine_adapter.DatabricksEngineAdapter]:
+        return engine_adapter.DatabricksEngineAdapter
 
     @property
-    def _connection_kwargs_keys(self) -> t.Set[str]:
-        if self.has_spark_session_access:
-            return set()
+    def _extra_engine_config(self) -> t.Dict[str, t.Any]:
         return {
-            "server_hostname",
-            "http_path",
-            "access_token",
-            "http_headers",
-            "session_configuration",
+            k: v
+            for k, v in self.dict().items()
+            if k.startswith("databricks_connect_") or k == "catalog"
         }
 
     @property
-    def _engine_adapter(self) -> t.Type[EngineAdapter]:
-        if self.has_spark_session_access:
-            return engine_adapter.DatabricksSparkSessionEngineAdapter
-        return engine_adapter.DatabricksSQLEngineAdapter
+    def use_spark_session_only(self) -> bool:
+        from sqlmesh import runtime_env
+
+        return runtime_env.is_databricks or self.force_databricks_connect
 
     @property
     def _connection_factory(self) -> t.Callable:
-        if self.has_spark_session_access:
+        if self.use_spark_session_only:
             from sqlmesh.engines.spark.db_api.spark_session import connection
 
             return connection
+
         from databricks import sql
 
         return sql.connect
 
     @property
     def _static_connection_kwargs(self) -> t.Dict[str, t.Any]:
-        if self.has_spark_session_access:
-            from pyspark import SparkConf
-            from pyspark.sql import SparkSession
+        from sqlmesh import runtime_env
+
+        if not self.use_spark_session_only:
+            return {}
 
-            spark_config = SparkConf()
-            if self.spark_config:
-                for k, v in self.spark_config.items():
-                    spark_config.set(k, v)
+        if runtime_env.is_databricks:
+            from pyspark.sql import SparkSession
 
             return dict(
-                spark=SparkSession.builder.config(conf=spark_config)
-                .enableHiveSupport()
-                .getOrCreate()
+                spark=SparkSession.builder.enableHiveSupport().getOrCreate(),
```

Comment:
Why are we building a spark session instead of getting an active one (`getActiveSession`)? If we do need to build it why do we no longer support custom spark config?

Reply:
Good point we don't need to build. Will change to `getActiveSession`. 

I removed the config because I'm not even sure if that worked within Databricks since the session already exists. Also I'm sure Databricks has constraints on what type of config you can set. So I decided to remove this and we can revisit if needed.